resources/rest-service/cloudify/migrations/versions/8e8314b1d848_6_2_to_6_3.py

```diff
@@ -360,3 +364,166 @@ def _add_deployment_resource_tags_column():
+            WHERE _count_users > max_users;
+            RETURN NULL;
+        END;
+    $$ LANGUAGE plpgsql; 
+    
+    CREATE FUNCTION increase_tenants_max() RETURNS TRIGGER AS $$
+        DECLARE
+            _count_tenants INTEGER;
+        BEGIN
+            SELECT COUNT(*) INTO _count_tenants FROM tenants; 
+            UPDATE usage_collector SET max_tenants = _count_tenants 
+            WHERE _count_tenants > max_tenants;
+            RETURN NULL;
+        END;
+    $$ LANGUAGE plpgsql; 
+    
+    CREATE FUNCTION increase_executions_total() RETURNS TRIGGER AS $$
+        BEGIN
+            UPDATE usage_collector SET total_executions = total_executions + 1;
+            RETURN NULL;
+        END;
+    $$ LANGUAGE plpgsql; 
+    
+    CREATE FUNCTION increase_logins_total() RETURNS TRIGGER AS $$
+        BEGIN
+            UPDATE usage_collector SET total_logins = total_logins + 1;
+            IF (OLD.last_login_at IS NULL)
+            AND NOT (NEW.last_login_at IS NULL) THEN
+                UPDATE usage_collector 
+                SET total_logged_in_users = total_logged_in_users + 1;
+            END IF;
+            RETURN NULL;
+        END;
+    $$ LANGUAGE plpgsql; 
+
+    CREATE TRIGGER increase_deployments_count 
+    AFTER INSERT ON deployments FOR EACH ROW
+    EXECUTE PROCEDURE increase_deployments_count();
+
+    CREATE TRIGGER increase_blueprints_count 
+    AFTER INSERT ON blueprints FOR EACH ROW
+    EXECUTE PROCEDURE increase_blueprints_count();
+    
+    CREATE TRIGGER increase_users_max 
+    AFTER INSERT ON users FOR EACH ROW
+    EXECUTE PROCEDURE increase_users_max();
+    
+    CREATE TRIGGER increase_tenants_max 
+    AFTER INSERT ON tenants FOR EACH ROW
+    EXECUTE PROCEDURE increase_tenants_max();
+    
+    CREATE TRIGGER increase_executions_total 
+    AFTER INSERT ON executions FOR EACH ROW
+    EXECUTE PROCEDURE increase_executions_total();
+    
+    CREATE TRIGGER increase_logins_total 
+    AFTER UPDATE OF last_login_at ON users FOR EACH ROW
+    EXECUTE PROCEDURE increase_logins_total();
+    """)
+
+
+def _drop_usage_collector_triggers():
+    op.execute("""DROP TRIGGER increase_deployments_count ON deployments;""")
+    op.execute("""DROP TRIGGER increase_blueprints_count ON blueprints;""")
+    op.execute("""DROP TRIGGER increase_users_max ON users;""")
+    op.execute("""DROP TRIGGER increase_tenants_max ON tenants;""")
+    op.execute("""DROP TRIGGER increase_executions_total ON executions;""")
+    op.execute("""DROP TRIGGER increase_logins_total ON users;""")
+
+    op.execute("""DROP FUNCTION increase_deployments_count();""")
+    op.execute("""DROP FUNCTION increase_blueprints_count();""")
+    op.execute("""DROP FUNCTION increase_users_max();""")
+    op.execute("""DROP FUNCTION increase_tenants_max();""")
+    op.execute("""DROP FUNCTION increase_executions_total();""")
+    op.execute("""DROP FUNCTION increase_logins_total();""")
```

Comment:
it's a bit ugly that they're all created in a single call but dropped in separate calls